### PR TITLE
feat(Docs): Docs refresh (Grid, Legacy Colors, Typography) [GM-118] [GM-194] 

### PR DESCRIPTION
**Pull request by codecaaron**
### Overview

These are the last backlogged PRs of docs updates that we haven't shipped yet.  These are mostly design guidelines and have less information about actual implementation, we point to theme for those.

These are unlikely to be final but they are more up to date than the current versions:
- Make the color documentation a bit more Color focused (grid view) and remove sections on SCSS once we have cut over.
- Add `themed` documentation to theme.

### Changes
- Removes Layout foundations documentation and moves it to `Grid` which is specific to top level layout.
- Updates `Color` and `Typography` to new style and refreshes some of the documentation comments.
- Removes our old Swatch component to use the DataTable component.

606b63a9a16ffc00a32e695d--gamut-preview netlify app_storybook__path=_docs_foundations-grid--page (1)
606b63a9a16ffc00a32e695d--gamut-preview netlify app_storybook__path=_docs_foundations-grid--page

### PR Checklist

- [x] Related to designs: [Grid](https://www.figma.com/file/ZUJ7cfVgyoh1kiQYNN8Hbt/Grid-%E2%9C%85?node-id=250%3A411) - [Typography](https://www.figma.com/file/rNsdbOwlw6L0ea3uAJrrz7/Typography?node-id=8%3A165)
- [x] Related to JIRA ticket: GM-118 GM-194
- [x] I have run this code to verify it works
- [ ] This PR includes unit tests for the code change

### CODE REVIEW

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/styleguide@33.6.3-alpha.036838.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://606b63a9a16ffc00a32e695d--gamut-preview.netlify.app>https://606b63a9a16ffc00a32e695d--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/606b63a9a16ffc00a32e695d>Deploy Logs</a></p>

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/styleguide@33.6.3-alpha.fa274b.0<br/></pre>

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/styleguide@33.6.3-alpha.a38fcf.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://606b690181af7d1145c375e4--gamut-preview.netlify.app>https://606b690181af7d1145c375e4--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/606b690181af7d1145c375e4>Deploy Logs</a></p>

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/styleguide@33.6.3-alpha.9bce58.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://606b6a767587a000825cb48a--gamut-preview.netlify.app>https://606b6a767587a000825cb48a--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/606b6a767587a000825cb48a>Deploy Logs</a></p>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://606b82a6f2ad14009e23f5eb--gamut-preview.netlify.app>https://606b82a6f2ad14009e23f5eb--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/606b82a6f2ad14009e23f5eb>Deploy Logs</a></p>

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/styleguide@33.6.3-alpha.16034a.0<br/></pre>

File: packages/styleguide/stories/Foundations/Colors/index.stories.mdx [hunk 1]
```diff
@@ -1,285 +1,106 @@
...
+    status: 'updating',
   }}
 />
 
 **Deprecation Notice** These variables and access patterns are now deprecated. Please refer to <LinkTo id="foundations-theme--colors">Theme</LinkTo> for the most up to date documentation when using the new visual identity.
 
-## Common Color Sets
+## Usage
+
+For the core color palette please use the emotion theme to access them.
```
Comment (dreamwasp) [inline packages/styleguide/stories/Foundations/Colors/index.stories.mdx:22]: maybe we could have a couple of examples here

File: packages/styleguide/stories/Foundations/Layout.stories.mdx [hunk 2]
```diff
@@ -1,155 +1,176 @@
...
+### Best Practices
 
-.container__medium {
-  max-width: $medium-content-width;
-}
-```
+- You should begin most design work with a target viewport of 1440px width, using the `XL` column grid.
+- Make sure you adjust each design to the viewports and take into account their respective screenheights (the ‘fold’).
+- Wider layouts should maintain proper line-lengths and component widths by adjusting arrangements to take advantage of the increased area—do not simply stretch to fill.
+- Elements that are not placed in an explicit lockup (like catalog cards) do not need to fill entire column widths. Alignment to gridlines on on axis (usually left side) is often enough
```
Comment (dreamwasp) [inline packages/styleguide/stories/Foundations/Layout.stories.mdx:108]: 🔥

**Review by dreamwasp — APPROVED** [review]
noted one small place where i think a code example would be useful but these changes are 🔥 and the documentation is super clear otherwise

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://606ca18ef65c0110994c56c8--gamut-preview.netlify.app>https://606ca18ef65c0110994c56c8--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/606ca18ef65c0110994c56c8>Deploy Logs</a></p>

**Pull request merged**